datafusion/common/src/scalar.rs

```diff
@@ -1653,42 +1619,66 @@ impl ScalarValue {
         Ok(array)
     }
 
-    fn iter_to_array_list(
-        scalars: impl IntoIterator<Item = ScalarValue>,
+    // This function does not contains nulls but empty array instead.
```

Comment:
NullBuffer is None in this function, you can't get `NULL` in the array from this function, only an empty array is possible.

I tried to have one function for `iter_to_array_list_without_nulls` and `iter_to_array_list` but failed since one needs nulls, and another needs an empty array.